tests/third_party/cupy/math_tests/test_sumprod.py

```diff
@@ -75,57 +74,57 @@ def test_sum_axis_huge(self, xp):
-        return xp.sum(a, **self._get_dtype_kwargs(xp, dtype), axis=1)
+        return xp.sum(a, **_get_dtype_kwargs(xp, dtype), axis=1)
 
     # float16 is omitted, since NumPy's sum on float16 arrays has more error
     # than CuPy's.
     @testing.for_all_dtypes(no_float16=True)
     @testing.numpy_cupy_allclose()
     def test_sum_axis2(self, xp, dtype):
         a = testing.shaped_arange((20, 30, 40), xp, dtype)
-        return a.sum(**self._get_dtype_kwargs(xp, dtype), axis=1)
+        return a.sum(**_get_dtype_kwargs(xp, dtype), axis=1)
 
     @testing.for_all_dtypes()
     @testing.numpy_cupy_allclose(contiguous_check=False)
     def test_sum_axis_transposed(self, xp, dtype):
         a = testing.shaped_arange((2, 3, 4), xp, dtype).transpose(2, 0, 1)
-        return a.sum(**self._get_dtype_kwargs(xp, dtype), axis=1)
+        return a.sum(**_get_dtype_kwargs(xp, dtype), axis=1)
 
     @testing.for_all_dtypes()
     @testing.numpy_cupy_allclose(contiguous_check=False)
     def test_sum_axis_transposed2(self, xp, dtype):
         a = testing.shaped_arange((20, 30, 40), xp, dtype).transpose(2, 0, 1)
-        return a.sum(**self._get_dtype_kwargs(xp, dtype), axis=1)
+        return a.sum(**_get_dtype_kwargs(xp, dtype), axis=1)
 
     @testing.for_all_dtypes()
     @testing.numpy_cupy_allclose()
     def test_sum_axes(self, xp, dtype):
         a = testing.shaped_arange((2, 3, 4, 5), xp, dtype)
-        return a.sum(**self._get_dtype_kwargs(xp, dtype), axis=(1, 3))
+        return a.sum(**_get_dtype_kwargs(xp, dtype), axis=(1, 3))
 
     @testing.for_all_dtypes()
     @testing.numpy_cupy_allclose(rtol=1e-4)
     def test_sum_axes2(self, xp, dtype):
         a = testing.shaped_arange((20, 30, 40, 50), xp, dtype)
-        return a.sum(**self._get_dtype_kwargs(xp, dtype), axis=(1, 3))
+        return a.sum(**_get_dtype_kwargs(xp, dtype), axis=(1, 3))
 
     @testing.for_all_dtypes()
     @testing.numpy_cupy_allclose(rtol=1e-6)
     def test_sum_axes3(self, xp, dtype):
         a = testing.shaped_arange((2, 3, 4, 5), xp, dtype)
-        return a.sum(**self._get_dtype_kwargs(xp, dtype), axis=(0, 2, 3))
+        return a.sum(**_get_dtype_kwargs(xp, dtype), axis=(0, 2, 3))
 
     @testing.for_all_dtypes()
     @testing.numpy_cupy_allclose(rtol=1e-6)
     def test_sum_axes4(self, xp, dtype):
         a = testing.shaped_arange((20, 30, 40, 50), xp, dtype)
-        return a.sum(**self._get_dtype_kwargs(xp, dtype), axis=(0, 2, 3))
+        return a.sum(**_get_dtype_kwargs(xp, dtype), axis=(0, 2, 3))
 
     @testing.for_all_dtypes()
     @testing.numpy_cupy_allclose()
     def test_sum_empty_axis(self, xp, dtype):
         a = testing.shaped_arange((2, 3, 4, 5), xp, dtype)
-        return a.sum(**self._get_dtype_kwargs(xp, dtype), axis=())
+        return a.sum(**_get_dtype_kwargs(xp, dtype), axis=())
 
     @testing.for_all_dtypes_combination(names=["src_dtype", "dst_dtype"])
     @testing.numpy_cupy_allclose()
```

Comment:
It is not needed since when `dtype` is determined for `sum`, the casting happens regardless it is safe or not.
But we can update the code to align with cupy.

Reply:
No, you are right, it looks fine to keep the difference with cupy here.